### PR TITLE
Add cr_ macros to set compiler flags, instead of using .bazelrc

### PR DESCRIPTION
**Pull request by chenhao94**
We enable some warnings/errors for our code, but we do not want to
affect the third-party libs, so macros may be a better choice.

Also, `-Wconversion` may be too strong, because it may be too verbose
to cast signed or unsigned integers everywhere. Wfloat-conversion may be
enough.

### CODE REVIEW

**Review by zhouxindong — APPROVED** [review]

File: BUILD [hunk 1]
```diff
@@ -1,16 +1,14 @@
 package(default_visibility = ["//visibility:public"])
 
-cc_library (
+load("//bazel:rules.bzl", "cr_cc_library")
```
Comment (xkszltl) [inline BUILD:3]: cr是指cris？干脆就写全吧

File: tests/timer_test.cc [hunk 2]
```diff
@@ -8,9 +8,11 @@
 namespace cris::core {
 
 template<class T1, class T2>
-void ExpectNear(const T1& v1, const T2& v2, float error = 0.05f) {
-    EXPECT_LT((1 - error) * v2, v1);
-    EXPECT_LT(v1, (1 + error) * v2);
+void ExpectNear(const T1& v1, const T2& v2, double error = 0.05) {
+    auto d1 = static_cast<double>(v1);
```
Comment (xkszltl) [inline tests/timer_test.cc:12]: const

File: bazel/rules.bzl [hunk 3]
```diff
@@ -0,0 +1,37 @@
+CRIS_CXX_COPTS = [
```
Comment (xkszltl) [inline bazel/rules.bzl:1]: CXXOPTS？

File: src/timer/timer_disable.cc [hunk 4]
```diff
@@ -57,15 +57,15 @@ TimerSession TimerSection::StartTimerSession() {
...
     return {};
 }
 
-TimerSection* TimerSection::SubSection(const std::string& name) {
+TimerSection* TimerSection::SubSection(const std::string& /* name */) {
     return this;
 }
 
-void TimerSection::ReportDurationNsec(cr_timestamp_nsec_t duration) {
+void TimerSection::ReportDurationNsec(cr_timestamp_nsec_t) {
```
Comment (xkszltl) [inline src/timer/timer_disable.cc:68]: 这个就直接给删了吗？

File: src/timer/timer_enable.cc [hunk 5]
```diff
@@ -347,14 +347,15 @@ cr_duration_nsec_t TimerReport::GetPercentileDurationNsec(int percent) const {
         percent = 100;
     }
 
-    const auto total_hits  = GetTotalHits();
-    const auto target_hits = std::round(total_hits * percent / 100.);
+    const auto total_hits = GetTotalHits();
+    const auto target_hits =
+        static_cast<unsigned long long>(std::llround(static_cast<double>(total_hits * percent) / 100.));
```
Comment (xkszltl) [inline src/timer/timer_enable.cc:352]: static_cast<unsigned long long>(round(total_hits * (pct / 100.)))

Comment (xkszltl) [reply src/timer/timer_enable.cc:352]: 或者你要是不怕爆掉的话原来那样就挺好，毕竟你去double走一圈丢掉的可不止7bit（100还不到7bit），double才50+bit的小数

**Review by xkszltl — CHANGES_REQUESTED** [review]

File: src/timer/timer_enable.cc (hunk 5 above)
Comment (chenhao94) [reply src/timer/timer_enable.cc:352]: 这里丢一点精度其实没影响，那我round好了

Comment (xkszltl) [reply src/timer/timer_enable.cc:352]: 只要你不是搞10000%这种输入，直接round的精度就比llround的写法高

File: src/timer/timer_disable.cc [hunk 6]
```diff
@@ -27,11 +27,11 @@ cr_duration_nsec_t TimerReport::GetAverageDurationNsec() const {
     return 0;
 }
 
-cr_duration_nsec_t TimerReport::GetPercentileDurationNsec(int percent) const {
+cr_duration_nsec_t TimerReport::GetPercentileDurationNsec(int /* percent */) const {
     return 0;
 }
 
-void TimerReport::PrintToLog(int indent_level) const {
+void TimerReport::PrintToLog(unsigned /* indent_level */) const {
```
Comment (xkszltl) [inline src/timer/timer_disable.cc:34]: const?

File: src/timer/timer_disable.cc [hunk 7]
```diff
@@ -57,15 +57,15 @@ TimerSession TimerSection::StartTimerSession() {
     return {};
 }
 
-std::unique_ptr<TimerReport> TimerSection::GetReport(bool recursive) {
+std::unique_ptr<TimerReport> TimerSection::GetReport(bool /* recursive */) {
```
Comment (xkszltl) [inline src/timer/timer_disable.cc:60]: const?

**Review by xkszltl — APPROVED** [review]

File: includes/message/lock_queue.h [hunk 8]
```diff
@@ -8,23 +8,23 @@ namespace cris::core {
...
     bool IsEmpty() override;
 
     bool IsFull() override;
 
    private:
-    const size_t                  capacity_;
-    size_t                        size_{0};
-    size_t                        begin_{0};
-    size_t                        end_{0};
+    const std::size_t             capacity_;
```
Comment (xkszltl) [inline includes/message/lock_queue.h:24]: init?

File: includes/node/single_queue_node.h [hunk 9]
```diff
@@ -22,8 +22,8 @@ class CRSingleQueueNode : public CRNode {
         return queues;
     }
 
-    size_t  queue_capacity_;
-    queue_t queue_;
+    std::size_t queue_capacity_;
```
Comment (xkszltl) [inline includes/node/single_queue_node.h:25]: init?

File: includes/node_runner/runner.h [hunk 10]
```diff
@@ -49,13 +49,13 @@ class CRMultiThreadNodeRunner : public CRNodeRunnerBase {
...
 
-    virtual std::function<void()> GetWorker(size_t thread_idx, size_t thread_num) = 0;
+    virtual std::function<void()> GetWorker(std::size_t thread_idx, std::size_t thread_num) = 0;
 
     virtual void NotifyWorkersToStop() = 0;
 
    private:
     CRNodeBase*              node_;
-    size_t                   thread_num_;
+    std::size_t              thread_num_;
```
Comment (xkszltl) [inline includes/node_runner/runner.h:58]: init?

**Pull request merged**